apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/McsClaimTransformerV2.java

```diff
@@ -318,19 +152,29 @@ private static List<Claim.DiagnosisComponent> getDiagnosis(RdaMcsClaim claimGrou
         .stream()
         .map(
             diagCode -> {
-              String icdVersion = diagCode.getIdrDiagIcdType().equals("0") ? "10" : "9-cm";
-
-              return new Claim.DiagnosisComponent()
-                  .setSequence(diagCode.getPriority() + 1)
-                  .setDiagnosis(
-                      new CodeableConcept()
-                          .setCoding(
-                              List.of(
-                                  new Coding(
-                                      "http://hl7.org/fhir/sid/icd-" + icdVersion,
-                                      diagCode.getIdrDiagCode(),
-                                      null))));
+              Claim.DiagnosisComponent component;
+
+              if (Strings.isNotBlank(diagCode.getIdrDiagCode())) {
+                String system;
+
+                if (VALID_ICD_TYPES.contains(diagCode.getIdrDiagIcdType())) {
+                  String icdVersion = diagCode.getIdrDiagIcdType().equals("0") ? "10" : "9-cm";
```

Comment:
@KyRobbins The system versions we should alternate between are `IcdCode.CODING_SYSTEM_ICD_9` and `IcdCode.CODING_SYSTEM_ICD_10_CM`